internal/shellgen/flake_input.go

```diff
@@ -54,6 +57,9 @@ func (f *flakeInput) BuildInputs() ([]string, error) {
 		if attributePathErr != nil {
 			err = attributePathErr
 		}
+		if pkg.PatchGlibc {
+			return strings.Replace(attributePath, "legacyPackages", "packages", 1)
```

Comment:
Sure. It's because the attribute path from nixpkgs is something like "legacyPackages.x86_64-linux.hello` but when it comes from the flake it's `packages.x86_64-linux.hello` (following the standard flake output schema).